ravenframework/Optimizers/fitness/fitness.py

```diff
@@ -139,23 +143,42 @@ def feasibleFirst(rlz,**kwargs):
           'constraintFunction', xr.Dataarray, containing all constraint functions (explicit and implicit) evaluations for the whole population
     @ Out, fitness, xr.DataArray, the fitness function of the given objective corresponding to a specific chromosome.
   """
-  objVar = kwargs['objVar']
-  g = kwargs['constraintFunction']
-  data = np.atleast_1d(rlz[objVar].data)
-  worstObj = max(data)
-  fitness = []
-  for ind in range(data.size):
-    if np.all(g.data[ind, :]>=0):
-      fit=(data[ind])
-    else:
-      fit = worstObj
-      for constInd,_ in enumerate(g['Constraint'].data):
-        fit+=(max(0,-1 * g.data[ind, constInd]))
-    fitness.append(-1 * fit)
-  fitness = xr.DataArray(np.array(fitness),
+  optType = kwargs['type']
+  objVar = [kwargs['objVar']] if isinstance(kwargs['objVar'], str) == True else kwargs['objVar']
+  a = 1.0                     if kwargs['a'] == None                       else kwargs['a']
+  if kwargs['constraintNum'] == 0:
+    pen = kwargs['b']
+  else:
+    g = kwargs['constraintFunction']
+    penalty = kwargs['b']
+    pen = [penalty[i:i+len(g['Constraint'].data)] for i in range(0, len(penalty), len(g['Constraint'].data))]
+  objPen = dict(map(lambda i,j : (i,j), objVar, pen))
+
+  for i in range(len(objVar)):
+    data = np.atleast_1d(rlz[objVar][objVar[i]].data)
+    worstObj = max(data)
+    fitness = []
+    for ind in range(data.size):
+      if kwargs['constraintNum'] == 0 or np.all(g.data[ind, :]>=0):
+        fit=(a*data[ind])
+      else:
```

Comment:
If you went with my suggestions about a being consistent with b. maybe this should be a[i] in line 163, and 165.

Reply:
This comment is related to the previous comment.